### ISSUE TITLE
pi05_libero_finetuned and pi05_libero_base get 0% success on LIBERO-10 with official lerobot-eval, while pi0_libero_finetuned works

### ISSUE THREAD

**Issue by jaavad** [huggingface/lerobot#3638]
### Ticket Type

🐛 Bug Report (Something isn't working)

### Environment & System Info

```Shell
(lerobot) javad@ZXP-S-works:~/projects/lerobot$ cd ~/projects/lerobot
(lerobot) javad@ZXP-S-works:~/projects/lerobot$ 
(lerobot) javad@ZXP-S-works:~/projects/lerobot$ git log --oneline -5
scribe --tags 2>/dev/null || true

python - <<'PY'
import torch
import transformers
import lerobot
0a4a7c40 (HEAD -> main, origin/main, origin/HEAD) docs(cheat sheet): create cheat sheet (#3602)
ca9028ad docs(quickstart): adding rollout  (#3598)
9db9c35c fix(config): add lora_alpha to PeftConfig (#3573)
fe96b28c Fix policy.path not working in YAML config files (#3145)
2438df13 chore(dependencies): update uv.lock (#3561)
(lerobot) javad@ZXP-S-works:~/projects/lerobot$ git describe --tags 2>/dev/null || true
rint("torch:", torch.__version__)
print("cuda available:", torch.cuda.is_available())
print("gpu:", v0.5.1-88-g0a4a7c40
(lerobot) javad@ZXP-S-works:~/projects/lerobot$ 
(lerobot) javad@ZXP-S-works:~/projects/lerobot$ python - <<'PY'
> import torch
> import transformers
> import lerobot
> print("torch:", torch.__version__)
> print("cuda available:", torch.cuda.is_available())
> print("gpu:", torch.cuda.get_device_name(0) if torch.cuda.is_available() else None)
> print("transformers:", transformers.__version__)
> print("lerobot:", getattr(lerobot, "__version__", "no __version__"))
> PY
torch: 2.12.0.dev20260407+cu128
cuda available: True
gpu: NVIDIA GeForce RTX 3090
transformers: 5.5.4
lerobot: 0.5.2
(lerobot) javad@ZXP-S-works:~/projects/lerobot$ 

(lerobot) javad@ZXP-S-works:~/projects/lerobot$ lerobot-info
- LeRobot version: 0.5.2
- Platform: Linux-5.15.0-139-generic-x86_64-with-glibc2.31
- Python version: 3.12.13
- Huggingface Hub version: 1.14.0
- Transformers version: 5.5.4
- Datasets version: N/A
- Numpy version: 2.2.6
- FFmpeg version: 8.1.1
- PyTorch version: 2.12.0.dev20260407+cu128
- Torchcodec version: N/A
- Is PyTorch built with CUDA support?: True
- Cuda version: 12.8
- GPU model: NVIDIA GeForce RTX 3090
- Using GPU in script?: <fill in>
- lerobot scripts: ['lerobot-calibrate', 'lerobot-dataset-viz', 'lerobot-edit-dataset', 'lerobot-eval', 'lerobot-find-cameras', 'lerobot-find-joint-limits', 'lerobot-find-port', 'lerobot-imgtransform-viz', 'lerobot-info', 'lerobot-record', 'lerobot-replay', 'lerobot-rollout', 'lerobot-setup-can', 'lerobot-setup-motors', 'lerobot-teleoperate', 'lerobot-train', 'lerobot-train-tokenizer']
(lerobot) javad@ZXP-S-works:~/projects/lerobot$
```

## Description

I am trying to evaluate `lerobot/pi05_libero_finetuned` on LIBERO-10 using the official `lerobot-eval` pipeline. However, the policy gets 0% success and the rollout videos show unintelligent / biased behavior.

This does not seem to be caused by my custom rollout code, because I reproduced the issue using the official `lerobot-eval` command directly.

I also tested `lerobot/pi0_libero_finetuned` in the same environment, and it behaves much more reasonably. In my run, the robot was able to perform meaningful task-directed behavior and picked up the tomato sauce in one episode. This suggests that my LIBERO installation, MuJoCo/EGL rendering, LeRobot environment wrapper, and action interface are working, and the issue may be specific to the `pi05_libero_finetuned` checkpoint or its loading/preprocessing compatibility.

## Command I ran

```bash
cd ~/projects/lerobot

MUJOCO_GL=egl CUDA_VISIBLE_DEVICES=1 lerobot-eval \
  --policy.path=lerobot/pi05_libero_finetuned \
  --env.type=libero \
  --env.task=libero_10 \
  --env.task_ids='[0]' \
  --eval.batch_size=1 \
  --eval.n_episodes=3 \
  --env.max_parallel_tasks=1 \
  --output_dir=./eval_logs/sanity_pi05_finetuned \
  2>&1 | tee /tmp/lerobot_eval.log

### Context & Reproduction

## Steps to reproduce

1. Use the current LeRobot main branch.
2. Run the official `lerobot-eval` command on LIBERO-10 task 0 with `lerobot/pi05_libero_finetuned`.
3. Watch the generated rollout videos and check the final success metric.

## Command I ran

```bash
cd ~/projects/lerobot

MUJOCO_GL=egl CUDA_VISIBLE_DEVICES=1 lerobot-eval \
  --policy.path=lerobot/pi05_libero_finetuned \
  --env.type=libero \
  --env.task=libero_10 \
  --env.task_ids='[0]' \
  --eval.batch_size=1 \
  --eval.n_episodes=3 \
  --env.max_parallel_tasks=1 \
  --output_dir=./eval_logs/sanity_pi05_finetuned \
  2>&1 | tee /tmp/lerobot_eval.log

Result

The final evaluation reports 0% success:

pc_success: 0.0
n_episodes: 3


### Relevant logs or stack trace

```Shell
There is no crash. The evaluation runs successfully, but the final success rate is 0%.

Relevant final result:


pc_success: 0.0
n_episodes: 3
```

### Checklist

- [ ] I have searched existing tickets to ensure this isn't a duplicate.
- [x] I am using the latest version of the `main` branch.
- [x] I have verified this is not an environment-specific problem.

## Additional information

I tested `lerobot/pi0_libero_finetuned` in the same environment, and it behaves much more reasonably. In my run, the robot performed meaningful task-directed behavior and picked up the tomato sauce in one episode.

This suggests that the LIBERO installation, MuJoCo/EGL rendering, LeRobot environment wrapper, and action interface are likely working.

## Additional observation from saved rollout videos

I compared the generated videos for three checkpoints in the same LIBERO setup:

- `lerobot/pi05_libero_finetuned`
- `lerobot/pi05_libero_base`
- `lerobot/pi0_libero_finetuned`

For both `pi05_libero_finetuned` and `pi05_libero_base`, the rollout videos show biased and non-task-directed behavior. The robot does not appear to make meaningful progress toward the instruction.

In contrast, `lerobot/pi0_libero_finetuned` behaves much more rationally in the same environment. The robot shows task-directed behavior and performs meaningful subtasks. For example, in one episode it picked up the tomato sauce.

This comparison is important because all three runs use the same machine, same LeRobot installation, same LIBERO environment, same MuJoCo/EGL setup, and the same official `lerobot-eval` pipeline. Therefore, the issue seems specific to the π0.5 checkpoints or their loading / preprocessing / normalization compatibility, rather than a general LIBERO environment, action-interface, or rendering problem.

## Possible causes

Could this be related to one of the following?

- checkpoint conversion issue
- normalization/statistics mismatch
- processor compatibility issue
- expected LeRobot version / commit mismatch
- `pi05_libero_finetuned` requiring a different checkpoint variant or older LeRobot commit

## Question

Could you clarify which LeRobot version / commit `lerobot/pi05_libero_finetuned` is expected to work with?

Is there a known issue with this checkpoint on the current main branch?

**Comment by pkooij** [comment]
Hi we did some changes to pi05, to improve the model and be truthful to original implementation regarding padding etc, https://github.com/huggingface/lerobot/commit/f0d2b37bebddf7e8852cb624712da0ad564601b6#diff-aba6d2e6b98327c579c2811c53c41444983e8122c140de5df462e75676f91d38. This could have caused degradation of finetuned models, such as one for libero. Finetuning it on libero would solve that issue, perhaps there are community checkpoints you could use out of the box.

**Comment by brisker** [comment]
@jaavad  have  you solved this?
